### DISCUSSION TITLE
不同的 CD 有可能抓取到完全一致的音频数据吗？

### DISCUSSION THREAD

**Discussion by giscus** [ZexWoo/Giscus#32] [Announcements]
# 不同的 CD 有可能抓取到完全一致的音频数据吗？

同专辑的两张 CD，抓轨都是 100%，则数据上是否存在差异？本文为你完全解析这个问题。

https://zexwoo.blog/posts/research/same-data-from-different-cds/

**Comment by dumbbird** [comment]
「前者不在乎偏移值，因此前述的情况 D 和 E，在 CTDB 处会拥有相同的 CTDB ID，由此，你在判断「版本」的时候，但凡两张碟 Log 末尾的 CTDB ID 相同，无需多想，绝对是同一个版本。」

对于这个结论，我可能要持保留意见了。我之前似乎见到过CTDBID相同，但两张专辑完全不一致的情况。找时间我去找找案例。如果是这样的话，CTDBID也只是一个参考而已，具体情况还是要具体分析。

**Reply by dumbbird** [reply]
案例：http://db.cuetools.net/top.php?tocid=.XqweGSDHMaxxZ4YUFFsSTJBhh4-

Artist	Album	Disc Id	Tracks	CTDB Id	Cf
未知艺术家	未知标题	.XqweGSDHMaxxZ4YUFFsSTJBhh4-	2	42AA5182	72
BOB MARLEY, MC LYTE	JAMMIN	.XqweGSDHMaxxZ4YUFFsSTJBhh4-	2	D2127119	2
周杰倫	黃金甲	.XqweGSDHMaxxZ4YUFFsSTJBhh4-	2	347D6FB2	1


这是周杰伦《黄金甲》EP碟。CueTools的CTDBID似乎只是根据音轨的时长信息来生成ID。所以像这种两三首歌的细碟EP，可能就会出现overlap的情况。两张EP 完全不搭噶，但是「恰好」时长完全相同，CT就会判定他们是同一个ID。但很显然，他们不是一个版本，甚至都不是同一张专辑。

**Reply by dumbbird** [reply]
为了避免误解，这里再解释一下。CueTools DB好像有两种不同的ID，一个是disc ID，是一长串字符，比如「.XqweGSDHMaxxZ4YUFFsSTJBhh4-」。通常这段字符被展现在抓轨log里。而这个disc ID 是无法证明两张专辑是同一版本的。这只是根据专辑音轨时长（TOC）来判定的。

而真正的CTDB ID，其实是这样的：「42AA5182」「D2127119」CRC 32的值。这个值才代表专辑的实际版本。如果这个值相同，那基本可以判定是同一版本。但问题在于这个值有时候并没有显示在抓轨log里面， cambia那个网站好像有时也无法给出这个信息。

**Reply by ZexWoo** [reply]
我亲自测试了一下，你说得没错，我会更新此文。感谢！
